src/Chemistry.hpp

```diff
@@ -74,7 +74,11 @@ template <typename problem_t> void computeChemistry(amrex::MultiFab &mf, const R
 
 			// do the actual integration
 			burner(chemstate, dt);
-
+#ifdef AMREX_USE_GPU
+			if (!chemstate.success) {
+				amrex::Abort("VODE integration was unsuccessful!");
+			}
+#endif
```

Comment:
Why did you add an `ifdef` here? I think this should abort in either case, unless there's something I'm missing?